flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/FactoryUtil.java

```diff
@@ -620,10 +624,16 @@ public static String getFormatPrefix(
             Class<T> factoryClass, DynamicTableFactory.Context context) {
         final String connectorOption = context.getCatalogTable().getOptions().get(CONNECTOR.key());
         if (connectorOption == null) {
-            throw new ValidationException(
-                    String.format(
-                            "Table options do not contain an option key '%s' for discovering a connector.",
-                            CONNECTOR.key()));
+            ManagedTableFactory factory =
+                    ManagedTableFactory.discoverManagedTableFactory(context.getClassLoader());
+            if (!factoryClass.isAssignableFrom(factory.getClass())) {
+                throw new ValidationException(
+                        String.format(
+                                "The managed table factory '%s' dose not implement '%s'.",
```

Comment:
the exception contains a typo and is not very nice. in the end we should have a helpful exception that includes the old exception:

```
Table options do not contain an option key '%s' for discovering a connector. Therefore, Flink assumes a managed table. However, a managed table factory is not in the classpath. 
```

Either we wrap the original exception or call some internal discoverFactory with an optional return type.

Reply:
Nice! I prefer to throw new exception directly, we don't need repetitive old exception.

Reply:
I created `FactoryUtil.discoverManagedTableFactory`, it throws exception messages for `not found` and `multiple found`.